src/click-outside.directive.ts

```diff
@@ -52,12 +54,16 @@ export class ClickOutsideDirective implements OnInit, OnChanges, OnDestroy {
     }
 
     this._events.forEach(e => document.body.removeEventListener(e, this._onClickBody));
+
+    if (this.emitOnBlur) {
```

Comment:
As previously mentioned, we should use a separate flag to track whether the listener was added or not so that we can be sure to *always* remove it after it's been added.

Reply:
Okay, done this way.